### PR TITLE
Cap toz

### PR DESCRIPTION
**Pull request by capTony02Zheng**
made baical structure of plot review to the left bar, including getting the chapter and showing the corresponding scenes, have not implement the case when remove current open project yet, fix the bug that prop not update with parent file

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -32,6 +37,14 @@ function setEditorMusic(event: MouseEvent) {
             task_id: projectID.value,
         });
 }
+
+function update_Preview_bar(event: MouseEvent) {
```
Comment (Tesseract22) [inline src/App.vue:41]: all function should use camelCase naming

File: src/RequestAPI.ts [hunk 2]
```diff
@@ -10,6 +11,10 @@ export enum Rtype {
     music = "music",
     character = "character",
 }
+export enum Frametype {
```
Comment (Tesseract22) [inline src/RequestAPI.ts:14]: I don't think an extra Frametype is neccessary. If an 'IFrame` has non-null `branch`, we know it is a `divergence` frame

File: src/RequestAPI.ts [hunk 3]
```diff
@@ -191,6 +196,45 @@ export async function renameResource(
...
+
+/**
+ * get all the frames of the corresponding chapter
+ * @param name: project_name
+ * @param chapter_name: chapter_name
+ * @returns
+ */
+export async function getFrames(name: string, chapter_name: string): Promise<IFrame[]> {
+    
+    let a,c: IFrame;
```
Comment (Tesseract22) [inline src/RequestAPI.ts:218]: `a,b,c` should be `const`. Also the name should be more descriptive. Single character name should not be allowed except for `iteration variable`

File: src/RequestAPI.ts [hunk 4]
```diff
@@ -191,6 +196,45 @@ export async function renameResource(
...
+    return ["test chapter","next is chapter name",name, "end of chapter" ];
+}
+
+/**
+ * get all the frames of the corresponding chapter
+ * @param name: project_name
+ * @param chapter_name: chapter_name
+ * @returns
+ */
+export async function getFrames(name: string, chapter_name: string): Promise<IFrame[]> {
```
Comment (Tesseract22) [inline src/RequestAPI.ts:216]: Consider adding error handling to make it more robust

File: src/components/ChapterDir.vue [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+import { ref, inject, watchEffect, type PropType } from "vue";
+import type { Ref } from "vue";
+import IconDownArrow from "./icons/IconDownArrow.vue";
+import type { IFrame } from "@/FrameDef";
+import { projectIDKey } from "../InjectionKeys";
+import { getFrames, Frametype } from "../RequestAPI"; // need to change to needed function
+
+var FramesDisplay = ref(false); // control display the scene of the corresopnding chapter
+const FrameList: Ref<IFrame[]> = ref([]);
+const chapname: Ref<string> = ref("");
```
Comment (Tesseract22) [inline src/components/ChapterDir.vue:12]: Use `chapName` instead

Comment (Tesseract22) [reply src/components/ChapterDir.vue:12]: Also, if you are on vscode, you easily rename a variable with `F2`

File: src/components/ChapterDir.vue [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+// function removeFrame(idx: Number): boolean {
+//     return false;
+// }
+
+// function switchBranch(id: FrameID): boolean {
+//     return false;
+// }
+
+const props = defineProps({
+    Chaptername: {
```
Comment (Tesseract22) [inline src/components/ChapterDir.vue:26]: Naming issue same as above

**Review by Tesseract22 — CHANGES_REQUESTED** [review]
Need some few changes in variable and function naming.

File: src/RequestAPI.ts (hunk 3 above)
Comment (capTony02Zheng) [reply src/RequestAPI.ts]: those were for testing, should be deleted in the future

File: src/RequestAPI.ts (hunk 4 above)
Comment (capTony02Zheng) [reply src/RequestAPI.ts:215]: waiting for back end to finish

**Pull request closed without merging**